optimum/habana/modeling_utils.py

```diff
@@ -38,6 +47,12 @@ def adapt_transformers_to_gaudi(use_habana_mixed_precision: bool):
     # Optimization tweak for ViT
     ViTSelfAttention.forward = gaudi_vit_self_attention_forward
 
+    # Optimization tweak for Wav2Vec2
+    _compute_mask_indices = _gaudi_wav2vec2_compute_mask_indices
+    _sample_negative_indices = _gaudi_wav2vec2_sample_negative_indices
+    Wav2Vec2Model._mask_hidden_states = _gaudi_wav2vec2_mask_hidden_states
+    Wav2Vec2Model.forward = gaudi_wav2vec2_forward
+
```

Comment:
why do we need gaudi specific change?

Reply:
To generate mask and negative indices directly on HPU. In Transformers, those are generated on CPU, which slows down the process. This is also done in your Model References.